### PR TITLE
Image comparison tests in CI

### PR DESCRIPTION
**Pull request by 1ucian0**
The image comparison testing can run in Azure, given that it does in the same env that binder.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -607,3 +607,39 @@ stages:
...
+            pip install -U -r requirements.txt -c constraints.txt
+            pip install -c constraints.txt -e .
+            pip install "matplotlib<3.3.0" pylatexenc pillow
+            python setup.py build_ext --inplace
+            sudo apt install -y graphviz pandoc
+            pip check
+          displayName: 'Install dependencies'
+        - bash: |
+            set -e
+            python -m unittest discover -v test.ipynb
```
Comment (mtreinish) [inline azure-pipelines.yml:644]:
```suggestion
        - bash: python -m unittest discover -v test/ipynb/mpl
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -607,3 +607,37 @@ stages:
...
+            python -m pip install --upgrade pip
+            pip install -U -r requirements.txt -c constraints.txt
+            pip install -c constraints.txt -e .
+            pip install "matplotlib<3.3.0" pylatexenc pillow
+            python setup.py build_ext --inplace
+            sudo apt install -y graphviz pandoc
+            pip check
+          displayName: 'Install dependencies'
+        - bash: python -m unittest discover -v test/ipynb
+          displayName: 'Run image test'
```
Comment (mtreinish) [inline azure-pipelines.yml:643]: Do you want to add an artifact upload here for the generated images? Also another option is we can use `jupyter nbconvert` to generate a pdf (or some other format) and upload that as artifacts so people can easily see the diffs.

**Review by mtreinish — APPROVED** [review]
LGTM, just one inline question

File: azure-pipelines.yml (hunk 2 above)
Comment (1ucian0) [reply azure-pipelines.yml:643]: Sure. I can add the zip creation in a follow up PR. I want to merge this ASAP to avoid the references to get out-of-sync.

Comment (1ucian0) [reply azure-pipelines.yml:643]: ahh! Maybe too late?

[the remark this answers is not in the document]
Comment (1ucian0) [reply azure-pipelines.yml:643]: dup https://github.com/Qiskit/qiskit-terra/pull/5108#discussion_r495148854

Comment (1ucian0) [reply azure-pipelines.yml:643]: oh.. hold on. There is something else going on here.

**Comment by 1ucian0** [comment]
Setting `on hold`, since it looks like the tests are failing when mybinder passes. https://mybinder.org/v2/gh/qiskit/qiskit-terra/master?urlpath=apps/test/ipynb/mpl_tester.ipynb

It needs to be debug. Any idea @mtreinish ?

File: azure-pipelines.yml (hunk 2 above)
Comment (1ucian0) [reply azure-pipelines.yml:643]: https://github.com/Qiskit/qiskit-terra/pull/4893 introduced a sneaky dependency

**Comment by 1ucian0** [comment]
https://github.com/Qiskit/qiskit-terra/pull/5136 blocks this issue. Therefore, `on hold`.

**Pull request merged**